Create User endpoints for logging in/out

adupuis2 opened the pull request:
For #11 

I just bundled registration into the login process since we lean on LDAP. Can/will modify to have a separate endpoint for registration if you think this is a bad move.

This is based off the `link_stored_procedures_to_python` branch. Putting it in draft mode until that one is merged.

adupuis2 committed:
Add User model with basic functionality
Implement callproc pass-through in database_helper
Add some basic endpoints to access and create User records

adupuis2 committed:
Create user endpoints and user sessions

adupuis2 commented on src/rickbox/ldap/ldap_helper.py:
```diff
@@ -49,11 +49,11 @@ def authenticate(self, username, password):
             connection.start_tls()
             connection.bind()
-            return 200, 'successfully authenticated user'
+            return 201, 'successfully authenticated user'
```
This was changed to bring it in line with the RAML specification.

jasonncleveland commented on src/rickbox/resources/user_session.py:
```diff
@@ -0,0 +1,65 @@
+        if not request.json \
+                or 'username' not in request.json \
+                or 'password' not in request.json:
+            return Response(json.dumps({'message': 'Bad request'}), 403)
```
Should this be a 400 since the input parameters are incorrect?

jasonncleveland commented on src/rickbox/app.py:
```diff
@@ -1,12 +1,42 @@
+api.add_resource(UserManager, '/users/<user_identifier>')
+
 
 @app.route('/')
```
This endpoint can be removed now.

jasonncleveland commented on src/rickbox/ldap/ldap_helper.py:
```diff
@@ -49,11 +49,11 @@ def authenticate(self, username, password):
-        except LDAPException:
-            return 500, 'an unexpected error occurred'
+        except LDAPException as e:
+            return 500, f'an unexpected error occurred: {e}'
```
This should probably be changed to log the error to the console instead of returning it as part of the string.

adupuis2 committed:
Change response code 403 -> 400 on invalid request to the log in handler
Print LDAP exception directly to console instead of returning it as part of the error string
